src/Utils/worker/cite.worker.ts

```diff
@@ -0,0 +1,149 @@
+import CSL, { MetaData as reference, Citation, CitationRegistry } from "citeproc";
+
+const ctx: Worker = self as any;
+let itemsObj = {};
+let style = null;
+let localesObj = {};
+let preferredLocale = null;
+let citeproc = null;
+let citationByIndex = null;
+let referenceData = [];
+
+interface CitationItem {
+  locator?: string;
+  label?: string;
+  "suppress-author"?: boolean;
+  "author-only"?: boolean;
+  prefix?: string;
+  suffix?: string;
+  id: string;
+}
+
+const sys = {
+  retrieveItem: function (itemID: string | number) {
+    return itemsObj[itemID];
+  },
+  retrieveLocale: function (lang: string) {
+    return localesObj[lang];
+  },
+};
+
+function getLocale(localeId: string) {
+  let xhr = new XMLHttpRequest();
```

Comment:
Can also be used with fetch? You probably need to set the headers ContentType to xml then